### ISSUE TITLE
Using only one CPU core

### ISSUE THREAD

**Issue by seventhsite** [iterativv/NostalgiaForInfinity#444]
Hi!
I use all default configs downloaded from this git in 3 fregtrade docker containers live. 
Sometimes I see when one or more processes use 100% of one processor core. 

![Screenshot from 2024-11-23 17-20-48](https://github.com/user-attachments/assets/da911d2d-dd5c-44cb-a117-05b5c213990e)

I don't see any errors, like _"Outdated history for pair..."_ or _"Strategy analysis took..."_ I just want to ask if this is normal behavior, that there is no parallelization across cores? Or should I think about switching to a more powerful processor with a higher frequency per core? Now I use Hetzner VPS on ARM Ampere Altra CPU.

**Comment by HeimLabor** [comment]
Yes, this is normal behavior. Freqtrade runs on a single core due to Python's Global Interpreter Lock (GIL). Computational intensive strategies, such as NostalgiaForInfinity, which rely on multiple indicators and historical data, can cause spikes to 100% on one core, especially during indicator calculations.

However, this typically has no significant impact because trading strategies like NostalgiaForInfinity don't rely on millisecond-level precision. The bot operates on candle intervals (e.g., 5 minutes), so slight delays during processing usually don't significantly affect trade execution or performance.

Switching to a CPU with higher single-core performance may help if the load causes noticeable delays, but as you mentioned, nothing appears in the error logs. I tested this on an i9-13900H and also observed occasional spikes to 100% on the utilized single core.
